sentry-android-core/src/main/java/io/sentry/android/core/SpanFrameMetricsCollector.java

```diff
@@ -68,63 +110,122 @@ public void onSpanFinished(final @NotNull ISpan span) {
     if (!enabled) {
       return;
     }
+
     if (span instanceof NoOpSpan) {
       return;
     }
+
     if (span instanceof NoOpTransaction) {
       return;
     }
 
```

Comment:
guess we should also ignore a span that never started collecting frames?
something like `if (!spanStarts.contains(span.getSpanContext().getSpanId().toString()) { return; }`